hydrus/hydraspec/openapi-hydra_parser.py

```diff
@@ -0,0 +1,113 @@
+"""
+Module to take in Open Api Specification and convert it to HYDRA Api Doc
+
+"""
+import yaml
+from doc_writer import HydraDoc, HydraClass, HydraClassProp, HydraClassOp
+import json
+
+with open("openapi_sample.yaml", 'r') as stream:
+    try:
+        doc = yaml.load(stream)
+    except yaml.YAMLError as exc:
+        print(exc)
+
+info = doc["info"]
+desc = info["description"]
+title = info["title"]
+baseURL = doc["host"]
+name = doc["basePath"]
+api_doc = HydraDoc(name, title, desc, name, baseURL)
+definitions = doc["definitions"]
+paths = doc["paths"]
+definitionSet = set()
+classAndClassDefinition = dict()
+hydra_doc = ""
+
+
+def getClasses():
+    """
+    Get Definitions from Open Api specification and convert to Classes and supported Props for Hydra Documentation
+    """
+    for class_ in definitions:
+        try:
+            desc = definitions[class_]["description"]
+        except KeyError:
+            desc = class_
+        classDefinition = HydraClass(class_, class_, desc, endpoint=False)
+        properties = definitions[class_]["properties"]
+        classAndClassDefinition[class_] = classDefinition
+        definitionSet.add(class_)
+        api_doc.add_supported_class(classDefinition, collection=False)
```

Comment:
How are you checking if any endpoint is a collection or not?

Reply:
I am not right now , in our discussions we decided to do that later .